lib/rspec/mocks/any_instance/expect_chain_chain.rb

```diff
@@ -23,11 +23,7 @@ def create_message_expectation_on(instance)
         end
 
         def invocation_order
-          @invocation_order ||= {
-            :and_return => [nil],
-            :and_raise => [nil],
-            :and_yield => [nil]
-          }
+          @invocation_order ||= {}
```

Comment:
Seems kindy silly to cache an empty hash.   And actually, all of the `invocation_order` methods return static hashes that are never different for different instances.  WDYT about refactoring to move them into a frozen constant?  If it helps implementation it could always keep the instance method (which would just return the constant) but that way we don't waste memory by creating new copies of the exact same hashes.